toolkit/tools/imagegen/configuration/networkconfig.go

```diff
@@ -0,0 +1,319 @@
+	// by dashes instead of colons. Therefore, we're reading from the 4th spot of the
+	// string to obtain the MAC address and then replace the dashes with colons
+	deviceAddr = strings.ReplaceAll(bootifValue[macAddressStartIndex:len(bootifValue)], "-", ":")
+	return
+}
+
+func checkNetworkDeviceAvailability(networkData Network) (deviceName string, err error) {
+	ifaces, err := net.Interfaces()
+	if err != nil {
+		return
+	}
+
+	if networkData.Device == "bootif" {
+		networkData.Device, err = findBootIfValue()
+		if err != nil {
+			logger.Log.Errorf("Failed to read bootif value from /proc/cmdline")
+			return
+		}
+	}
+
+	for _, iface := range ifaces {
+		ifaceHardwareAddr := strings.TrimSpace(iface.HardwareAddr.String())
+		if networkData.Device == iface.Name || strings.EqualFold(ifaceHardwareAddr, networkData.Device) {
+			deviceName = iface.Name
+			return
+		}
+	}
+
+	return
+}
+
+func populateMatchSection(networkData Network, fileName, deviceName string) (err error) {
+	matchSection := fmt.Sprintf("[Match]\nName=%s\n", deviceName)
+	err = file.Append(matchSection+"\n", fileName)
+	if err != nil {
+		logger.Log.Errorf("Failed to write [Match] section: %s", err)
+	}
+
+	return
+}
+
+func populateNetworkSection(networkData Network, fileName string) (err error) {
+	var networkSection strings.Builder
+
+	// Write network section tag
+	networkSection.WriteString("[Network]\n")
+
+	// Currently only supports static IP setting
+	// Write IP and netmask
+	if networkData.NetMask != "" && networkData.Ip != "" {
+		stringMask := net.IPMask(net.ParseIP(networkData.NetMask).To4())
+		cidrNumber, _ := stringMask.Size()
+		networkSection.WriteString(fmt.Sprintf("Address=%s/%d\n", networkData.Ip, cidrNumber))
+	}
+
+	// Write Gateway
+	if networkData.GateWay != "" {
+		networkSection.WriteString(fmt.Sprintf("Gateway=%s\n", networkData.GateWay))
+	}
+
+	// Write Nameserver
+	for _, nameserver := range networkData.NameServers {
+		networkSection.WriteString(fmt.Sprintf("DNS=%s\n", nameserver))
+	}
+
+	err = file.Append(networkSection.String(), fileName)
+	if err != nil {
+		logger.Log.Errorf("Failed to write [Network] section: %s", err)
+	}
+
+	return
+}
+
+func createNetworkConfigFile(installChroot *safechroot.Chroot, networkData Network, deviceName string) (err error) {
+	const (
+		networkFileDir = "/etc/systemd/network"
+		filePrefix     = "10"
+		squashErrors   = false
+	)
+
+	if exists, ferr := file.DirExists(networkFileDir); ferr != nil {
+		logger.Log.Errorf("Error accessing /etc/systemd/network/")
+		err = ferr
+		return
+	} else if !exists {
+		err = fmt.Errorf("/etc/systemd/network/: no such path or directory")
+		return
+	}
+
+	logger.Log.Debugf("Start creating network file")
+
+	networkFilePath := fmt.Sprintf("%s/%s-%s-%s.network", networkFileDir, filePrefix, networkData.BootProto, deviceName)
+	exists, err := file.PathExists(networkFilePath)
+	if exists && err == nil {
```

Comment:
>if exists && err == nil { [](http://example.com/codeflow?start=1&length=25)

Please also check the value of `err`.

Reply:
I took another look and I believe in this specific case, there's actually no need to check the error code since the logic here is reversed. The error condition for this case is actually when err is equal to nil, because when err == nil, that means the network file path that we're going to create already exists in the system and thus raise an error. And the boolean returned by PathExists is also set by comparing err against nil, thus I think just checking the value of the boolean is enough.

Reply:
Fixed